tests/testsuite.cpp

```diff
@@ -19,24 +19,53 @@
 #include "testese.h"
 #include <assert.h>
 #include "config.h"
+#include <string>
+
+void check_nal_file(const char* uri, int log_level, ESEVideoCodec codec, std::string codec_name, int num_packets_nal, int num_packets_au)
+{
+
+  ESExtractor* extractor;
+
+  extractor = create_es_extractor(uri, nullptr, log_level);
+  assert(extractor);
+  assert(es_extractor_video_format(extractor) == ESE_VIDEO_FORMAT_NAL);
+  assert(es_extractor_video_codec(extractor) == codec);
+  assert(std::string(es_extractor_video_codec_name(extractor)) == codec_name);
+  assert(parse (extractor) == num_packets_nal);
+  es_extractor_set_options (extractor, "alignment:AU");
+  assert(parse (extractor) == num_packets_au);
+  es_extractor_set_options (extractor, "alignment:NAL");
+  assert(parse (extractor) == num_packets_nal);
+  es_extractor_teardown(extractor);
+}
+
+void check_ivf_file (const char* uri, int log_level, ESEVideoCodec codec, std::string codec_name, int num_packets)
+{
+  ESExtractor* extractor;
+
+  extractor = create_es_extractor(uri, nullptr, log_level);
+  assert(extractor);
+  assert(es_extractor_video_format(extractor) == ESE_VIDEO_FORMAT_IVF);
+  assert(es_extractor_video_codec(extractor) == codec);
+  assert(std::string(es_extractor_video_codec_name(extractor)) == codec_name);
```

Comment:
Ditto, unnecessary ctor?

Reply:
if I want  to "==" es_extractor_video_codec_name(extractor) I need a std::string ctor